src/conversions/num_bigint.rs

```diff
@@ -138,53 +140,69 @@ macro_rules! bigint_conversion {
-
-                    let n_bytes = {
+                    let n_bits = {
                         cfg_if::cfg_if! {
                             if #[cfg(not(Py_LIMITED_API))] {
                                 // fast path
                                 let n_bits = ffi::_PyLong_NumBits(num.as_ptr());
                                 if n_bits == (-1isize as usize) {
                                     return Err(crate::PyErr::fetch(py));
-                                } else if n_bits == 0 {
-                                    0
-                                } else {
-                                    (n_bits - 1 + $is_signed) / 8 + 1
                                 }
+                                n_bits
                             } else {
                                 // slow path
                                 let n_bits_obj = num.getattr(py, crate::intern!(py, "bit_length"))?.call0(py)?;
                                 let n_bits_int: &PyLong = n_bits_obj.downcast_unchecked(py);
-                                let n_bits = n_bits_int.extract::<usize>()?;
-                                if n_bits == 0 {
-                                    0
-                                } else {
-                                    (n_bits - 1 + $is_signed) / 8 + 1
-                                }
+                                n_bits_int.extract::<usize>()?
                             }
                         }
                     };
 
-                    if n_bytes <= 128 {
-                        let mut buffer = [0; 128];
-                        extract(num.as_ref(py), &mut buffer[..n_bytes], $is_signed)?;
-                        Ok($from_bytes(&buffer[..n_bytes]))
-                    } else {
-                        let mut buffer = vec![0; n_bytes];
-                        extract(num.as_ref(py), &mut buffer, $is_signed)?;
-                        Ok($from_bytes(&buffer))
+                    if n_bits == 0 {
+                        return Ok(0.try_into().unwrap())
                     }
+
+                    let n_digits = (n_bits + 31 + $is_signed) / 32;
+
+                    let mut buffer = Vec::<u32>::with_capacity(n_digits);
+                    buffer.set_len(n_digits);
+                    extract(
+                        num.as_ref(py),
+                        &mut buffer,
+                        $is_signed
+                    )?;
+
+                    #[cfg(target_endian = "big")]
+                    buffer
+                        .iter_mut()
+                        .for_each(|chunk|{*chunk = u32::from_le(*chunk)});
+
+                    Ok($from_digits(buffer))
                 }
             }
         }
     };
 }
-bigint_conversion!(BigUint, 0, BigUint::to_bytes_le, BigUint::from_bytes_le);
+bigint_conversion!(
+    BigUint,
+    0,
+    BigUint::to_bytes_le,
+    //from digits conversion
+    |digits: Vec<u32>| { BigUint::new(digits) }
```

Comment:
I think you don't need the closure here

```suggestion
    BigUint::new
```

Reply:
i just assumed BigUint::new doesnt copy data, but num_bigint::u32_to_u64 function is clearly doing it, and in a funny way as well. So now im just confused about the speedup, mostly...

Reply:
however, now im dreaming of the day num_bigint gets fixed and the speedup is massive! i think i know my next PR